packages/backend/src/open_payments/payment/outgoing/lifecycle.ts

```diff
@@ -158,6 +164,244 @@ function getAdjustedAmounts(
+
+  // Get the latest spent amounts record for this specific payment,
+  // not necessarily the latest for this grant
+  const latestPaymentSpentAmounts =
+    await OutgoingPaymentGrantSpentAmounts.query(deps.knex)
+      .where('outgoingPaymentId', payment.id)
+      .orderBy('createdAt', 'desc')
+      .first()
+
+  // TODO: this shouldnt happen. should we error instead?
+  if (!latestPaymentSpentAmounts) {
+    deps.logger.warn(
+      { outgoingPaymentId: payment.id },
+      'No outgoingPaymentGrantSpentAmounts record found for outgoingPaymentId'
+    )
+    return
+  }
+
+  // Detect if partial payment
+  // TODO: can partial payments be detected without this grant spent record?
+  // Like just from the payment vs. settledAmounts? If so we could remove the latestPaymentSpentAmounts query
+  const reservedDebitAmount = latestPaymentSpentAmounts.paymentDebitAmountValue
+  const reservedReceiveAmount =
+    latestPaymentSpentAmounts.paymentReceiveAmountValue
+  const debitAmountDifference = reservedDebitAmount - settledAmounts.debit
+  const receiveAmountDifference = reservedReceiveAmount - settledAmounts.receive
+
+  if (debitAmountDifference === 0n && receiveAmountDifference === 0n) return
+
+  // Get the latest spent amounts for this grant (all time)
+  const latestGrantSpentAmounts = await OutgoingPaymentGrantSpentAmounts.query(
+    deps.knex
+  )
+    .where('grantId', payment.grantId)
+    .orderBy('createdAt', 'desc')
+    .first()
+
+  if (!latestGrantSpentAmounts) {
+    deps.logger.warn(
+      { grantId: payment.grantId },
+      'No outgoingPaymentGrantSpentAmounts record found for grantId'
+    )
+    return
+  }
+
+  const newGrantTotalDebitAmountValue =
+    latestGrantSpentAmounts.grantTotalDebitAmountValue - debitAmountDifference
+  const newGrantTotalReceiveAmountValue =
+    latestGrantSpentAmounts.grantTotalReceiveAmountValue -
+    receiveAmountDifference
+
+  // For interval amounts, we need the latest record from this payment's interval
+  // (not necessarily the latest overall, nor this specific payment's spent amount record)
+  let newIntervalDebitAmountValue: bigint | null = null
+  let newIntervalReceiveAmountValue: bigint | null = null
+
+  if (latestPaymentSpentAmounts.intervalStart !== null) {
+    const latestIntervalSpentAmounts =
+      await OutgoingPaymentGrantSpentAmounts.query(deps.knex)
+        .where('grantId', payment.grantId)
+        .where('intervalStart', latestPaymentSpentAmounts.intervalStart)
+        .where('intervalEnd', latestPaymentSpentAmounts.intervalEnd)
+        .orderBy('createdAt', 'desc')
+        .first()
+
+    if (!latestIntervalSpentAmounts) {
+      deps.logger.warn(
+        {
+          grantId: payment.grantId,
+          intervalStart: latestPaymentSpentAmounts.intervalStart,
+          intervalEnd: latestPaymentSpentAmounts.intervalEnd
+        },
+        'No outgoingPaymentGrantSpentAmounts record found for grant interval'
+      )
+      return
```

Comment:
Need to think through this a bit, but I think we would still want to update the `OutgoingPaymentGrantSpentAmounts` instead of returning early

Reply:
Like, as a fallback? Is that your thinking?

I guess maybe that makes sense. This should never happen though so IDK how we'd get the correct interval/grant totals. Since it's totally unexpected I dont think we can just assume the interval/grant amounts were 0 or something. I suppose we can sum them all up like we do for the legacy path on the payment create side.